lib/vim-state.coffee

```diff
@@ -661,15 +650,26 @@ class VimState
     text = @getRegister(name)?.text
     @editor.insertText(text) if text?
 
-  ensureCursorIsWithinLine: (cursor) =>
-    return if @processing or @mode isnt 'normal'
+  # Private: ensure the mode follows the state of selections
+  checkSelections: =>
+    return unless @editor?
+    if @editor.getSelections().every((selection) -> selection.isEmpty())
+      @ensureCursorsWithinLine() if @mode is 'normal'
```

Comment:
quite right; in fact, now it's the check in `ensureCursorIsWithinLine` that's no longer necessary so I removed that one instead.